### PR TITLE
update xgboost functions

### PR DESCRIPTION
**Pull request by double-y**
### Description
Made xgboost_reg, xgboost_binary, xgboost_multi for build_model and separated tidier methods for them

resolve https://github.com/exploratory-io/tam/issues/4752

### Checklist

Make sure you have performed following items before submitting this pull request.
If not, please describe the reason.  

- [x] Add test cases for this fix/enhancement
- [x] Pass devtools::check()
- [x] Pass devtools::test()
- [ ] Test installing from github
- [x] Tested with Exploratory

### CODE REVIEW

File: R/util.R [hunk 1]
```diff
@@ -265,7 +265,13 @@ mat_to_df <- function(mat, cnames=NULL, na.rm=TRUE, diag=TRUE){
 #' match the type of two vector
 same_type <- function(vector, original){
   if(is.factor(original)){
-    as.factor(vector)
+    if(all(vector[!is.na(vector)] %in% levels(original))){
+      # if original is factor and vector has all values,
+      # should return factor with same levels
+      factor(vector, levels = vector)
```
Comment (hideaki) [inline R/util.R]: Did you mean something like this?
factor(vector, levels = levels(original))

File: R/build_xgboost.R [hunk 2]
```diff
@@ -96,53 +20,144 @@ fml_xgboost <- function(data, formula, weight = NULL, params = list(), ...) {
...
+  } else {
+    factored <- as.factor(data[[y_name]])
+    y_vals <- as.numeric(factored) - 1
+    same_type(levels(factored), data[[y_name]])
+  }
+
+  data[[y_name]] <- y_vals
+
+  objective <- paste0("multi:", output_type, sep = "")
+  ret <- ret <- data_xgboost(data = data,
```
Comment (hideaki) [inline R/build_xgboost.R]: why is ret substituted twice?

Comment (double-y) [inline R/build_xgboost.R]: It's my mistake. Will remove it

File: R/util.R (hunk 1 above)
Comment (double-y) [inline R/util.R]: Yes, it's mistake. Added test case and fixed

**Pull request merged**